src/linux/backend-v4l2.h

```diff
@@ -285,13 +285,42 @@ namespace librealsense
             std::queue<sync_buffer> _md_queue;
         };
 
+        struct v4l2_device_info
+        {
+            enum connection_type
+            {
+                CONNECTION_TYPE_USB = 0,
+                CONNECTION_TYPE_MIPI  = 1
+            };
+
+            std::string card;
+            std::string bus_info;
+            std::vector<std::string> video_nodes;
+
+            connection_type connection_from_bus_info(const std::string& bus_info);
+        };
+
         class v4l_uvc_device : public uvc_device, public v4l_uvc_interface
         {
         public:
             static void foreach_uvc_device(
                     std::function<void(const uvc_device_info&,
                                        const std::string&)> action);
 
+            static std::vector<std::string> get_video_paths();
+
+            static bool is_usb_path_valid(const std::string& usb_video_path, const std::string &dev_name,
+                                          std::string &busnum, std::string &devnum, std::string &devpath);
+
+            static bool is_usb_device_path(const std::string& video_path);
+
+            static uvc_device_info handle_usb_device_path(const std::string& video_path, const std::string& name);
```

Comment:
This 2 functions names should give some clew about their usage.
Maybe `get_info_from_usb_device_path` ? and same with mipi?

Reply:
I agree, I will correct